### PR TITLE
config: add storage for tikv config

### LINKED ISSUE #4811: pd will loss the tikv config  after pd restarted or transferred
## Bug Report
tikv config will miss after pd restarted or transferred, so merge will be failed.

### What did you do?
1. set tikv config: Coprocessor.MaxRegionSize: 10GB and restart.
2. restart pd
3. pd loss the tikv config 

### What did you expect to see?
pd will save the tikv config after pd restart. 
### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What problem does this PR solve?

Issue Number: Close #4811

### What is changed and how it works?
use etcd  storage to save the config of the tikv

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Code changes
- Has persistent data change

Side effects

Related changes

### Release note

```release-note
 None.
```

### CODE REVIEW

File: server/storage/endpoint/key_path.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ import (
 const (
 	clusterPath                = "raft"
 	configPath                 = "config"
+	storeConfigPath            = "store-config"
```
Comment (JmPotato) [inline server/storage/endpoint/key_path.go:25]:
```suggestion
	storeConfigPath            = "store_config"
```

File: server/statistics/store_collection.go [hunk 2]
```diff
@@ -238,10 +244,10 @@ type storeStatisticsMap struct {
 }
 
 // NewStoreStatisticsMap creates a new storeStatisticsMap.
-func NewStoreStatisticsMap(opt *config.PersistOptions) *storeStatisticsMap {
+func NewStoreStatisticsMap(opt *config.PersistOptions, manager *config.StoreConfigManager) *storeStatisticsMap {
```
Comment (rleungx) [inline server/statistics/store_collection.go:247]: How about only passing `*config.StoreConfig`?

File: server/config/store_config_test.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ package config
 import (
 	"encoding/json"
 	"fmt"
+	"github.com/tikv/pd/server/storage"
```
Comment (lhy1024) [inline server/config/store_config_test.go:20]: add an empty line?

File: server/storage/endpoint/key_path.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ import (
 const (
 	clusterPath                = "raft"
 	configPath                 = "config"
+	storeConfigPath            = "store_config"
```
Comment (lhy1024) [inline server/storage/endpoint/key_path.go:25]: need to put a more suitable path？

File: server/config/store_config_test.go (hunk 3 above)
Comment (bufferflies) [reply server/config/store_config_test.go:20]: fix

File: tests/server/member/member_test.go [hunk 5]
```diff
@@ -229,6 +229,36 @@ func (s *memberTestSuite) waitEtcdLeaderChange(c *C, server *tests.TestServer, o
...
+	c.Assert(storeConfigManager.GetStoreConfig().GetRegionMaxSize(), Equals, uint64(144))
+	storeConfigManager.UpdateConfig(&config.StoreConfig{
+		Coprocessor: config.Coprocessor{
+			RegionMaxSize: "10Gib",
+		},
+	}, storage)
+	storeConfigManager.Reload(storage)
+	c.Assert(storeConfigManager.GetStoreConfig().GetRegionMaxSize(), Equals, uint64(10*1024))
+
+	//
```
[the remark this answers is not in the document]
Comment (bufferflies) [reply tests/server/member/member_test.go]: fix

File: server/statistics/store_collection.go (hunk 2 above)
Comment (bufferflies) [reply server/statistics/store_collection.go:247]: In past,  the  point of the `*config.StoreConfig` will be replace not the value after config be updated. I will change it in next pr.

File: server/storage/endpoint/key_path.go (hunk 4 above)
Comment (bufferflies) [reply server/storage/endpoint/key_path.go]: fix

File: server/storage/endpoint/config.go [hunk 6]
```diff
@@ -74,3 +76,25 @@ func (se *StorageEndpoint) SaveScheduleConfig(scheduleName string, data []byte)
 func (se *StorageEndpoint) RemoveScheduleConfig(scheduleName string) error {
 	return se.Remove(scheduleConfigPath(scheduleName))
 }
+
+// LoadStoreConfig loads the config of store.
+func (se *StorageEndpoint) LoadStoreConfig(cfg interface{}) (bool, error) {
```
Comment (nolouch) [inline server/storage/endpoint/config.go:81]: Can it not persist? TiKV as a source of truth.

Comment (bufferflies) [reply server/storage/endpoint/config.go:81]: In past, the trigger of the updated config is `PutStore`, and this api will not be called if pd restart or transfer leader.

**Comment by bufferflies** [comment]
it will be closed because pd should save the config of tikv.

**Pull request closed without merging**